### ISSUE TITLE
please add an option to disable/enable the option dynamically

### ISSUE THREAD

**Issue by putchi** [elmarquis/Leaflet.GestureHandling#31]
please add an option to enable or disable the gestureHandling dynamically for example when entering full-screen.

this.map. gestureHandling.enabled(); // return bool
this.map. gestureHandling.disable();
this.map. gestureHandling.enable();

**Comment by elmarquis** [comment]
Sure. here you go. 
It's pretty much exactly as you posted. I actually used this same approach for a full screen implementation myself on a recent project. 

```
// Initialize the map as normal
var map = L.map("map", {
    center: [-25.2702, 134.2798],
    zoom: 3,
    gestureHandling: true,
    gestureHandlingOptions: {
        duration: 5000 //5 secs
    }
});

function goFullScreen() {
  map.gestureHandling.disable();
 // .. do the other stuff to launch full screen
}

function exitFullScreen() {
    map.gestureHandling.enable();
   // .. do other stuff to exit full screen
}

```

If you want to check if it's enabled, this works too. 

```
  map.gestureHandling.enabled(); //returns true/false
```

**Comment by tuckergordon** [comment]
Is this in the latest release? I'm not seeing that functionality implemented anywhere in the [source](https://github.com/elmarquis/Leaflet.GestureHandling/blob/master/src/js/leaflet-gesture-handling.js) and have been unable to get it working in my project thus far.

**Comment by putchi** [comment]
> Is this in the latest release? I'm not seeing that functionality implemented anywhere in the [source](https://github.com/elmarquis/Leaflet.GestureHandling/blob/master/src/js/leaflet-gesture-handling.js) and have been unable to get it working in my project thus far.

Yes, it is as @elmarquis said, it is already implemented but, it is not pat of the [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped). 

@elmarquis could you do a PR with the GestureHandling property type for the Map module Similar to [this one](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/leaflet-fullscreen)?

**Comment by tuckergordon** [comment]
My apologies, I forgot that those functions were in `L.Handler`. However I'm running into a bug with this still. If I do the following, `gestureHandling` the event handlers still appear to be there although `myMap.gestureHandling.enabled()` returns `false`.

```
const myMap = L.map("map", {
      gestureHandling: true
    });
myMap.gestureHandling.disable();
```

However, doing the opposite _does_ work:

```
const myMap = L.map("map", {
      gestureHandling: false
    });
myMap.gestureHandling.enable();
```

So for now my work around is just to do `enable()` and then `disable()`, which gives me the desired result. Haven't figured out where the bug is coming from yet.

**Comment by elmarquis** [comment]
Hi @tuckergordon 

I've just been testing it out on the example page and it's working ok for me. Might be I'm missing something though. 

https://elmarquis.github.io/Leaflet.GestureHandling/examples/

It's running code pretty similar to yours. 

```
var map = L.map("map", {
        center: [-25.2702, 134.2798],
        zoom: 3,
        gestureHandling: true
    });
```

Open the dev tools and do these commands in the console: 
```
map.gestureHandling.enabled() // returns true. Gesture handling working

map.gestureHandling.disable() //Disable it

map.gestureHandling.enabled() // returns false. Gesture removed from map. Appears to be functioning as normal. 
```

You were saying the event handlers still seem to be in place even though it's disabled. Do you get the whole dark overlay with text prompting you to scroll, even though it says it's disabled? I didn't experience this, but keen to know if you did and any other factors that might help me recreate.

**Comment by francoisromain** [comment]
Hello I use it with leaflet full screen plugin with the following code: 

```
     this.map.on('fullscreenchange', () => {
        if (this.map.isFullscreen()) {
          this.map.gestureHandling.disable()
          console.log('entered fullscreen', this.map.gestureHandling.enabled())
        } else {
          this.map.gestureHandling.enable()
          console.log('exited fullscreen', this.map.gestureHandling.enabled())
        }
      })
```

The logs are correct (`gestureHandling.enabled` is false in fullscreen), but gestureHandling is still enabled in fullscreen mode (with the dark overlay)

**Comment by francoisromain** [comment]
There is no bug if we don't use Javascript modules. 

The bug only occurs if we use modules. 

I made a bug repro here: https://github.com/francoisromain/bug-leaflet-gesture-handling

You can see the bug here: https://francoisromain.github.io/bug-leaflet-gesture-handling/

Thank you

**Comment by francoisromain** [comment]
actually, if we remove this line `L.Map.addInitHook("addHandler", "gestureHandling", GestureHandling);` everything works as expected… 

Is this necessary to add this line when we use the plugin as amodule?

**Comment by andy-UKC** [comment]
This bug still seems to be around, any idea of a workaround?
